suzieq/poller/controller/source/nautobot.py

```diff
@@ -0,0 +1,228 @@
+"""Nautobot inventory"""
+
+import asyncio
+import logging
+from typing import Dict, List, Optional, Union
+from urllib.parse import urlparse
+
+from pynautobot import api
+
+from suzieq.poller.controller.source.base_source import Source, SourceModel
+from suzieq.poller.controller.inventory_async_plugin import InventoryAsyncPlugin
+from suzieq.shared.utils import get_sensitive_data
+from suzieq.shared.exceptions import InventorySourceError, SensitiveLoadError
+
+# pylint: disable=no-name-in-module
+# pylint: disable=no-self-argument
+
+from pydantic import BaseModel, validator, Field
+
+logger = logging.getLogger(__name__)
+
+_DEFAULT_PORTS = {"http": 80, "https": 443}
+
+
+class NautobotServerModel(BaseModel):
+    """Model containing data to connect with Nautobot server"""
+
+    host: str
+    protocol: str
+    port: str
+
+    class Config:
+        """pydantic configuration"""
+
+        extra = "forbid"
+
+
+class NautobotSourceModel(SourceModel):
+    """Nautobot source validation model"""
+
+    period: Optional[int] = Field(default=3600)
+    token: str
+    ssl_verify: Optional[bool] = Field(alias="ssl-verify")
+    server: Union[str, NautobotServerModel] = Field(alias="url")
+    run_once: Optional[bool] = Field(default=False, alias="run_once")
+    device_filters: Optional[Dict] = Field(default=None, alias="device_filters")
```

Comment:
I don't see much validation on the `device_filters`. What happens if the user does the following?
``` yaml
- name: test
  type: nautobot
  device_filters:
    foo: bar
    bar: foo
```

Reply:
Pynautobot will catch the error and an `InventorySourceError` will be raised with message `ERROR - nautobot-instance-123: error while getting devices: The request failed with code 400 Bad Request: {'foo': ['Unknown filter field']}`. I added a test for this as well.